service/httpCall_test.go

```diff
@@ -0,0 +1,512 @@
+package service
+
+import (
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/kotanetes/go-test-it/model"
+)
+
+type mockClient struct {
+	DoReturn *http.Response
+	DoError  error
+	DoFn     func(*http.Request) (*http.Response, error)
+}
+
+func (m *mockClient) Do(req *http.Request) (*http.Response, error) {
+	if m.DoFn != nil {
+		return m.DoFn(req)
+	}
+	return m.DoReturn, m.DoError
+}
+
+func Test_CompareData(t *testing.T) {
+	t.Run("should return true for valid map data", func(t *testing.T) {
+		var result, expected map[string]interface{}
+
+		result = map[string]interface{}{
+			"id":              "16",
+			"employee_name":   "Michael Silva",
+			"employee_salary": "198500",
+			"employee_age":    "66",
+			"profile_image":   "",
+		}
+
+		expected = map[string]interface{}{
+			"id":              "16",
+			"employee_name":   "Michael Silva",
+			"employee_salary": "198500",
+			"employee_age":    "66",
+			"profile_image":   "",
+		}
+
+		final := true
+
+		r := compareData(t.Name(), result, expected)
+		if r != final {
+			t.Errorf("Test %v failed, expected not equeal to got", t.Name())
+		}
+	})
+
+	t.Run("should return true for the data of slice of interface", func(t *testing.T) {
+		var result, expected map[string]interface{}
+
+		result = map[string]interface{}{
+			"id":              "16",
+			"employee_name":   "Michael Silva",
+			"employee_salary": "198500",
+			"employee_age":    "66",
+			"additional_data": []interface{}{
+				map[string]interface{}{"address": "line 1"},
+				map[string]interface{}{"phone": "12345556"},
+			},
+		}
+
+		expected = map[string]interface{}{
+			"id":              "16",
+			"employee_name":   "Michael Silva",
+			"employee_salary": "198500",
+			"employee_age":    "66",
+			"additional_data": []interface{}{
+				map[string]interface{}{"address": "line 1"},
+				map[string]interface{}{"phone": "12345556"},
+			},
+		}
+
+		final := true
+
+		r := compareData(t.Name(), result, expected)
```

Comment:
I think here no need to declare the ```final``` variable because of ```compareData``` function will return either ```true``` or ```false```. Need to compare like this,
```
if !r {
			t.Errorf("%v failed, expected not equal to got", t.Name())
		}
```

Reply:
Yes, I agree